rcl/test/rcl/test_time.cpp

```diff
@@ -289,8 +289,8 @@ TEST(CLASSNAME(rcl_time, RMW_IMPLEMENTATION), specific_clock_instantiation) {
 
 TEST(CLASSNAME(rcl_time, RMW_IMPLEMENTATION), rcl_time_difference) {
   rcl_allocator_t allocator = rcl_get_default_allocator();
-  rcl_clock_t * ros_clock =
-    reinterpret_cast<rcl_clock_t *>(allocator.allocate(sizeof(rcl_clock_t), allocator.state));
+  auto ros_clock =
```

Comment:
I think there's a missing `auto *` on this line, but that's not going to cause the CI failures.

Reply:
I didn't think that auto needed the * since I thought it would infer it's correct type from the static cast. Is that not the case?

Reply:
> I didn't think that auto needed the * since I thought it would infer it's correct type from the static cast

This is indeed the case -- but, as you stated, it is inconsistent with other instances. Honestly, I am not a fan of this part of the `auto` keyword's behavior...